### PR TITLE
Fix Type Annotation in pandas.core.accessor

### LINKED ISSUE #26146: Fix Type Annotation in pandas/core/accessor.py
Part of #25882 

Current errors are:

    pandas\core\accessor.py:14: error: Need type annotation for '_accessors'

PR coming soon

### PR DESCRIPTION
**Pull request by vaibhavhrt**
- [x] closes #26146
- [x] tests passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by WillAyd** [comment]
Can you double check remaining modules while at it to make sure this is current?

**Comment by vaibhavhrt** [comment]
Ok, I will all the modules in mypy.ini, but that will take some time. I will remove all unnecessary modules from mypy.ini by tomorrow, or add a comment here if there are none.

File: pandas/core/accessor.py [hunk 1]
```diff
@@ -5,15 +5,17 @@
...
 from pandas.util._decorators import Appender
 
 
 class DirNamesMixin:
-    _accessors = frozenset()
+    _accessors = set()  # type: Set[str]
     _deprecations = frozenset(
-        ['asobject', 'base', 'data', 'flags', 'itemsize', 'strides'])
+        ['asobject', 'base', 'data', 'flags', 'itemsize', 'strides']
+    )  # type: FrozenSet[str]
```
Comment (WillAyd) [inline pandas/core/accessor.py]: Is this annotation required? I figured it should be able to infer from data and didn't see it listed in the linked issue

File: pandas/core/accessor.py [hunk 2]
```diff
@@ -5,15 +5,17 @@
...
 """
+from typing import FrozenSet, Set
 import warnings
 
 from pandas.util._decorators import Appender
 
 
 class DirNamesMixin:
-    _accessors = frozenset()
+    _accessors = set()  # type: Set[str]
```
Comment (jreback) [inline pandas/core/accessor.py]: why did you change this?

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/core/accessor.py (hunk 1 above)
Comment (vaibhavhrt) [reply pandas/core/accessor.py]: No it's not required, but `_deprecations` is typed everywhere(generic.py and frames.py) so I typed it here too.

File: pandas/core/accessor.py (hunk 2 above)
Comment (vaibhavhrt) [reply pandas/core/accessor.py]: In `generic.py` class `NDFrame` is grandchild/child of the `DirNamesMixin`. And in class `NDFrame` and `DataFrame`, `_accessors` is a `set`, so mypy will throw an error if I leave it to be a `frozenset` here.See this PR #25909 . More specifically start reading from here https://github.com/pandas-dev/pandas/pull/25909#issuecomment-477592965

Comment (jreback) [reply pandas/core/accessor.py]: ok that's fine

File: pandas/core/accessor.py (hunk 1 above)
Comment (WillAyd) [reply pandas/core/accessor.py]: Yea I understand your reasoning but I think would rather remove. Should leverage inference where correct and leave annotations to where required or inference is wrong.

**Review by WillAyd — CHANGES_REQUESTED** [review]
Comment on annotation

File: pandas/core/accessor.py (hunk 1 above)
Comment (vaibhavhrt) [reply pandas/core/accessor.py]: @WillAyd I will remove it hold on a minute.

**Comment by vaibhavhrt** [comment]
> 
> 
> Comment on annotation

@WillAyd Comment what on what. Sorry I didn't get what you actually mean.

> Can you double check remaining modules while at it to make sure this is current?

I am still checking the modules in `mypy.ini` for any unnecessary modules. I will be done tomorrow.

**Comment by vaibhavhrt** [comment]
@WillAyd I found few more unnecessary modules in `mypy.ini`. I have commented them out so that you can take a look and also to make sure tests pass. I will remove them after tests pass.

**Comment by WillAyd** [comment]
@vaibhavhrt feel free to remove and fix up lint issue

**Comment by vaibhavhrt** [comment]
@WillAyd all done. @jreback you can go ahead and merge this PR of everything looks ok to you.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @vaibhavhrt
